### PR TITLE
Runner refactoring

### PR DESCRIPTION
**Pull request by p1003**
.

### CODE REVIEW

File: pymchelper/executor/runner.py [hunk 1]
```diff
@@ -105,78 +112,79 @@ def get_data(output_dir):
...
-            DEVNULL = open(os.devnull, 'wb')
-            subprocess.check_call(command_as_list, cwd=workspace, stdout=DEVNULL, stderr=DEVNULL)
-
-        # in case of Ctrl-C
-        except KeyboardInterrupt:
-            logging.debug("KeyboardInterrupt")
-            raise KeyboardInterruptError()
-
-        return workspace
+    class Executor:
```
Comment (grzanka) [inline pymchelper/executor/runner.py:128]: Explain the reason why executor becomes inner class of the runner class ?

**Review by grzanka — APPROVED** [review]
Looks ok

File: pymchelper/executor/runner.py [hunk 2]
```diff
@@ -48,50 +40,65 @@ def __init__(self, jobs=None, settings=None):
...
         # SHIELD-HIT12A and Fluka require RNG seeds to be integers greater or equal to 1
         # each of the workers needs to have its own different RNG seed
         rng_seeds = range(1, self.jobs + 1)
 
+        # create working directories
+        self.workspace_manager.create_working_directories(simulation_input_path=settings.input_path,
+                                                          rng_seeds=rng_seeds)
+
+        # temporary rework of rng_seeds injection to settings (formerly executor was responsible for it)
+        # TODO rework it somehow   # skipcq: PYL-W0511
```
Comment (p1003) [inline pymchelper/executor/runner.py:64]:
```suggestion
        # rng seeds injection to settings for each SingleSimulationExecutor call
        # TODO consider better way of doing it  # skipcq: PYL-W0511
```

**Review by p1003 — COMMENTED** [review]
1 suggestion

**Pull request merged**